packages/smarthr-ui/src/components/Table/ThCheckbox.tsx

```diff
@@ -1,50 +1,59 @@
 import React, { forwardRef, useMemo } from 'react'
 import { tv } from 'tailwind-variants'
 
+import { Balloon } from '../Balloon'
 import { CheckBox, Props as CheckBoxProps } from '../CheckBox'
 import { Center } from '../Layout'
-import { VisuallyHiddenText } from '../VisuallyHiddenText'
 
 import { Th } from './Th'
 
 import type { DecoratorsType } from '../../types'
 
 type Props = {
-  decorators?: DecoratorsType<'checkAllInvisibleLabel'>
+  decorators?: DecoratorsType<'checkAllInvisibleLabel'> & {
+    checkColumnName?: (text: string) => string
+  }
 }
 
-const CHECK_ALL_INVISIBLE_LABEL = 'すべての行を選択'
+const CHECK_ALL_INVISIBLE_LABEL = 'すべての項目を選択/解除'
+const CHECK_COLUMN_NAME = '選択'
 
 const thCheckbox = tv({
   slots: {
-    inner: 'shr-absolute shr-inset-0 [&:not(:has([disabled]))]:shr-cursor-pointer',
+    inner: 'shr-absolute shr-inset-0 [&:not(:has([disabled]))]:shr-cursor-pointer shr-group',
     wrapper: 'shr-relative shr-w-[theme(fontSize.base)] [&]:shr-px-0.75',
     checkbox: '[&]:shr-block',
+    balloon:
+      '[&]:shr-absolute shr-left-[38px] shr-translate-y-[1px] shr-sr-only group-has-[:hover,:focus-visible]:shr-not-sr-only group-has-[:hover,:focus-visible]:shr-absolute group-has-[:hover,:focus-visible]:shr-whitespace-nowrap',
```

Comment:
マジックナンバー気になる。
38px を calc で出すか、コメント残すかしたい。

```suggestion
      '[&]:shr-absolute shr-left-[38px] shr-translate-y-px shr-sr-only group-has-[:hover,:focus-visible]:shr-not-sr-only group-has-[:hover,:focus-visible]:shr-absolute group-has-[:hover,:focus-visible]:shr-whitespace-nowrap',
```